Bottom: 668ddd4271538fc243a888fad972f12f0e1bf29d
Top:    2c19f63f87fca4a6072bdf2f255fb935db04cff0
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2013-08-22 11:30:33 -0700

cmtime: Add ability to time rdma_bind_addr calls

Signed-off-by: Sean Hefty <sean.hefty@intel.com>


---

diff --git a/examples/cmtime.c b/examples/cmtime.c
index f97ab9e..5b416af 100644
--- a/examples/cmtime.c
+++ b/examples/cmtime.c
@@ -54,6 +54,7 @@ static char *src_addr;
 
 enum step {
 	STEP_CREATE_ID,
+	STEP_BIND,
 	STEP_RESOLVE_ADDR,
 	STEP_RESOLVE_ROUTE,
 	STEP_CREATE_QP,
@@ -65,6 +66,7 @@ enum step {
 
 static char *step_str[] = {
 	"create id",
+	"bind addr",
 	"resolve addr",
 	"resolve route",
 	"create qp",
@@ -348,6 +350,22 @@ static int run_client(void)
 	conn_param.private_data = rai->ai_connect;
 	conn_param.private_data_len = rai->ai_connect_len;
 
+	if (src_addr) {
+		printf("binding source address");
+		start_time(STEP_BIND);
+		for (i = 0; i < connections; i++) {
+			start_perf(&nodes[i], STEP_BIND);
+			ret = rdma_bind_addr(nodes[i].id, rai->ai_src_addr);
+			if (ret) {
+				perror("failure bind addr");
+				nodes[i].error = 1;
+				continue;
+			}
+			stop_perf(&nodes[i], STEP_RESOLVE_ADDR);
+		}
+		end_time(STEP_BIND);
+	}
+
 	printf("resolving address\n");
 	start_time(STEP_RESOLVE_ADDR);
 	for (i = 0; i < connections; i++) {
